content/helpers.go

```diff
@@ -87,6 +89,81 @@ func WriteBlob(ctx context.Context, cs Ingester, ref string, r io.Reader, desc o
 	return Copy(ctx, cw, r, desc.Size, desc.Digest, opts...)
 }
 
+// WriteBlobLeased writes data with the expected digest into the content store. If
+// expected already exists, the method returns immediately and the reader will
+// not be consumed. It also adds the object as a reference to a lease.
```

Comment:
The backend already does this when the lease is provided through the context. If there was an issue discovered with that it would be a bug, but this extra function should not exist.